libr/util/file.c

```diff
@@ -1276,29 +1276,41 @@ R_API bool r_file_copy(const char *src, const char *dst) {
 }
 
 R_API bool r_file_dir_recursive(RList *dst, const char *dir) {
-	bool ret = false;
+	bool ret = true;
 	char *cwd = r_sys_getdir ();
 	if (!cwd) {
 		return false;
 	}
 	if (r_sys_chdir (dir) == false) {
 		free (cwd);
-		return ret;
+		return false;
 	}
 	RList *files = r_sys_dir (".");
 	RListIter *iter;
 	char *name;
 	r_return_val_if_fail (files, false);
```

Comment:
ONLY use r_return_* at the begining of functions to check function arguments. dont use that to chk what function calls return